interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/FragmentParameterValidator.scala

```diff
@@ -0,0 +1,57 @@
+package pl.touk.nussknacker.engine.definition
+
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.context.{ProcessCompilationError, ValidationContext}
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{
+  InitialValueNotPresentInPossibleValues,
+  RequireValueFromEmptyFixedList
+}
+import pl.touk.nussknacker.engine.compile.nodecompilation.NodeCompiler
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.engine.graph.node.FragmentInputDefinition.{
+  FragmentParameter,
+  FragmentParameterFixedValuesUserDefinedList,
+  FragmentParameterInputMode,
+  FragmentParameterNoFixedValues
+}
+
+object FragmentParameterValidator {
+
+  def validate(
+      fragmentParameter: FragmentParameter,
+      fragmentInputId: String,
+      compiler: NodeCompiler,
+      validationContext: ValidationContext // localVariables must include this and other FragmentParameters
+  )(implicit nodeId: NodeId): List[ProcessCompilationError] = {
+    val fixedValuesList = fragmentParameter match {
+      case _: FragmentParameterNoFixedValues              => None
+      case f: FragmentParameterFixedValuesUserDefinedList => Some(f.fixedValuesList)
+    }
+
+    val fixedValueResponses = (fixedValuesList.getOrElse(List.empty) ++ fragmentParameter.initialValue).map {
+      fixedExpressionValue =>
+        compiler.compileExpression(
+          expr = Expression.spel(fixedExpressionValue.expression),
+          ctx = validationContext,
+          expectedType = validationContext(fragmentParameter.name),
+          fieldName = fragmentParameter.name,
+          outputVar = None
+        )
+    }
+
+    val allowOnlyValuesFromFixedList =
+      fragmentParameter.inputMode == FragmentParameterInputMode.InputModeFixedList
+
+    List(
+      (allowOnlyValuesFromFixedList && fixedValuesList.isEmpty)
```

Comment:
What if parameter is not required? Or maybe `allowOnlyValuesFromFixedList` implies that `required == true`?

Reply:
InputModeFixedList means there is no way for the user to pick anything but one of the provided values, so no way to provide even empty spel expression (unless it's one of the values) or leave it undefined

I guess there's a case, where the parameter is not required, and the creator enforces a InputModeFixedList from an empty list to make sure that the parameter is never provided any value, but what's the point of such a parameter? :thinking:

Reply:
`there is no way for the user to pick anything but one of the provided values`
If the user uses GUI, right? :D Using API, user can send any value, right? ;)

Reply:
yeah, but then it will fail validation (unless it's a lazy value, but that's a different beast, we don't currently check if lazy values pass our validators)

Reply:
I guess we could prohibit lazy values from outside the fixed values list if inputMode == InputModeFixedList

but that's a question of design

Reply:
Ok, if it will fail later - that is fine by me. :)